### PR TITLE
Add entropy

### PR DESCRIPTION
**Pull request by Garishtech**
The score thresholds in generateFeedback() are still up to tweaking. This is currently without bonus bits, which can be as much as 32 without a dictionary check. 
Also up to change what suggestions to include based on scoring. Entropy is mainly based on length and charset size, so I'm not sure what other relevant suggestions could be given

EDIT: Found a more reliable source expressing the same bit thresholds for strength classifications: http://iocane.com.au/talking-passwords-and-entropy/#:~:text=The%20strength%20of%20a%20password,as%20well%20as%20password%20length.&text=in%20the%20password%20itself.

### CODE REVIEW

File: src/checkers/entropy.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    if(!result.number){description.push("Use numbers");}
+
+    //Scoring sorta from this website: https://www.pleacher.com/mp/mlessons/algebra/entropy.html
+    //<14 = 1
+    //<28 = very weak = 2-3
+    //28-35 weak = 4-5
+    //36-59 reasonable  = 6-7
+    //60-127 strong = 8-9
+    //128+ very strong = 10
+
```
Comment (Garishtech) [inline src/checkers/entropy.js:122]: Using these scores, "password" comes out as a strength of 6, which didn't really make sense so I tried to make up my own thresholds below. If we include bonus bits, these will need to be raised quite a bit, so it's undecided if those will be used

File: src/checkers/entropy.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                else{
+                    bonus += (length - 8) * 1.5;
+                }
+            }
+            else{
+                bonus += (length-1) * 2.0;
+            }
+        }
+    }
+*/
```
Comment (Garishtech) [inline src/checkers/entropy.js:51]: Bonus bits for length. Still not sure if we should use them.

File: src/checkers/entropy.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    if(special){charsetSize += 33;}
+    if(number){charsetSize += 10;}
+
+    //entropy formula according to NIST Eletronic Authentication Guidelines
+    if(charsetSize > 0){
+        entropy = Math.log2(charsetSize) * length;
+    } else{ 
+        entropy = 0;
+    }
+
```
Comment (Garishtech) [inline src/checkers/entropy.js:88]: The base entropy calculation. Would return "NaN" without the else statement.

File: src/checkers/entropy.js (hunk 1 above)
Comment (jemisonf) [reply src/checkers/entropy.js:122]: FWIW if we followed the algorithm correctly and it's doing stuff like ranking "password" as a 6 that's actually fine, that helps show the strengths/weaknesses of the different approaches

**Pull request merged**